api/handlers.go

```diff
@@ -0,0 +1,31 @@
+package api
+
+import (
+	"fmt"
+	"github.com/ONSdigital/dp-sessions-api/decoder"
+	. "github.com/ONSdigital/dp-sessions-api/session"
+	"github.com/ONSdigital/log.go/log"
+	"net/http"
+)
+
+// CreateSessionHandlerFunc returns a function that generates a session. Method = "POST"
+func CreateSessionHandlerFunc() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		var c CreateSessionEntity
+		if err := decoder.Decode(r, &c); err != nil {
+			log.Event(ctx, "missing a required field", log.Error(err), log.ERROR)
+			http.Error(w, "Missing a required field", http.StatusBadRequest)
+			return
+		}
+
+		sess := CreateNewSession(c.Email)
```

Comment:
As discussed:

Created a new interface type defined here in the handler package which will be responsible for the sessions logic
```
type SessionService interface {
    CreateNewSession(email string) session.Session {
}
```
Update the create handler function to take an argument of type `SessionService`
```
CreateSessionHandlerFunc(sessionService SessionService)
```

Use the session service to create the new session when handling a request
```
sessionService.CreateNewSession(c.Email)
```
This decouples the `handler` and `session` packages and gives you more control. You can now provided the implementation you want use when you create the handler. For the app this will be a real implementation for the test you will use a mock.